nilearn/datasets/atlas.py

```diff
@@ -429,6 +429,139 @@ def fetch_atlas_harvard_oxford(atlas_name, data_dir=None,
+    See documentation for details.
+
+    Parameters
+    ----------
+    atlas_name : string
+        Name of atlas to load. Can be:
+        maxprob-thr0-1mm,  maxprob-thr0-2mm,
+        maxprob-thr25-1mm, maxprob-thr25-2mm,
+        maxprob-thr50-1mm, maxprob-thr50-2mm,
+        prob-1mm,          prob-2mm
+
+    data_dir : string, optional
+        Path of data directory where data will be stored. Optionally,
+        it can also be a FSL installation directory (which is dependent
+        on your installation).
+        Example, if FSL is installed in /usr/share/fsl/ then
+        specifying as '/usr/share/' can get you Juelich Histological atlas
+        from your installed directory. Since we mimic same root directory
+        as FSL to load it easily from your installation.
+
+    symmetric_split : bool, optional
+        If True, lateralized atlases will be returned. 
+        Default=False.
+
+    resume : bool, optional
+        Whether to resumed download of a partly-downloaded file.
+        Default=True.
+
+    verbose : int, optional
+        Verbosity level (0 means no message). Default=1.
+
+    Returns
+    -------
+    data : sklearn.datasets.base.Bunch
+        Dictionary-like object, keys are:
+
+        - "maps": nibabel.Nifti1Image, 4D maps if a probabilistic atlas is
+          requested and 3D labels if a maximum probabilistic atlas was
+          requested.
+
+        - "labels": string list, labels of the regions in the atlas.
+
+    """
+    atlas_items = ("maxprob-thr0-1mm", "maxprob-thr0-2mm",
+                   "maxprob-thr25-1mm", "maxprob-thr25-2mm",
+                   "maxprob-thr50-1mm", "maxprob-thr50-2mm",
+                   "prob-1mm", "prob-2mm")
+    if atlas_name not in atlas_items:
+        raise ValueError("Invalid atlas name: {0}. Please chose an atlas "
+                         "among:\n{1}".format(
+                             atlas_name, '\n'.join(atlas_items)))
+
+    url = "https://cdn.filesend.jp/private/1McfSSXL0-f67vqFrnEIbQZbfvS7YHn8ZTHoDL657Goj2FZFMnbsOp21SE1QpdYO/Juelich.tgz"
+
+    # For practical reasons, we mimic the FSL data directory here.
+    dataset_name = 'fsl'
+    data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir,
+                                verbose=verbose)
+
+    opts = {'uncompress': True}
+    root = os.path.join('data', 'atlases')
+
+    label_file = "Juelich.xml"
+    label_file = os.path.join(root, label_file)
+
+    atlas_file = os.path.join(root, 'Juelich',
+                              'Juelich-' + atlas_name + '.nii.gz')
+    
+    atlas_img, label_file = _fetch_files(
+        data_dir,
+        [(atlas_file, url, opts), (label_file, url, opts)],
+        resume=resume, verbose=verbose)
+
+    names = {}
+    from xml.etree import ElementTree
+    names[0] = 'Background'
+    for label in ElementTree.parse(label_file).findall('.//label'):
+        names[int(label.get('index')) + 1] = label.text
+    names = list(names.values())
```

Comment:
the order is the order of insertion (not, for example, of the sorted keys). why use a dict and take the values rather than a list from the start? are there duplicate indexes?

Reply:
if the goal is to sort by index please use `names = [item[1] for item in sorted(names.items())]`